### PR TITLE
Add header for fixing build warnings of file partitions.c

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
Add header file for building partitions.c, to clean up build warnings.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [99dc579aeb9abe6c9b9e6da0b4c7d72304191960] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [99dc579aeb9abe6c9b9e6da0b4c7d72304191960] - Code Rule Check Result:
os/board/esp32_devkit/src/esp32_boot.c:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: os/board/common/partitions.c [hunk 1]
```diff
@@ -25,6 +25,8 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (sunghan-chang) [inline os/board/common/partitions.c]: Could you add this next to config.h?
```
#include <tinyara/config.h>

#include <stdio.h>
#include <stdlib.h>
#include <debug.h>
...
```

**Comment by seinfra** [comment]
Target : [118b3676b236950cadb06612675286505b861f8e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [118b3676b236950cadb06612675286505b861f8e] - Code Rule Check Result:
os/board/esp32_devkit/src/esp32_boot.c:229: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

File: os/board/common/partitions.c (hunk 1 above)
Comment (ChinaTizenRT) [reply os/board/common/partitions.c]: fixed

**Comment by seinfra** [comment]
Target : [8e7753491f61703ebc5b2bde96be34478427c890] - Code Rule Check OK.

File: os/board/common/partitions.c (hunk 1 above)
Comment (sunghan-chang) [reply os/board/common/partitions.c]: I don't know what you fixed. I mean I want to position them next of config.h as shown above. Anyway I merge this now.

**Comment by seinfra** [comment]
Target : [8e7753491f61703ebc5b2bde96be34478427c890] - Code Rule Check (C++) OK.

**Pull request merged**